pkg/controller/notary/controller.go

```diff
@@ -84,97 +84,110 @@ func (c *Controller) mutatePodSpec(namespace, specPath string, pod corev1.PodSpe
-				a.StringToAdmissionResponse(fmt.Sprintf("Trust Server/Image Configuration Error: %v", err.Error()))
+			// Make sure image sure there is a ImagePullSecret defined
+			// TODO: This prevents use of signed publically available images with publically available signing data
+			var registryToken string
+			if len(pod.ImagePullSecrets) == 0 {
+				a.StringToAdmissionResponse(fmt.Sprintf("Deny %q, no ImagePullSecret defined for %s", img.String(), img.GetHostname()))
 				continue containerLoop
 			}
-			digest, err := c.getDigest(notaryURL, img.NameWithoutTag(), notaryToken, img.GetTag(), signers)
-			if err != nil {
-				if strings.Contains(err.Error(), "401") {
+
+		secretLoop:
+			for _, secret := range pod.ImagePullSecrets {
+				registryToken, err = c.kubeClientsetWrapper.GetSecretToken(namespace, secret.Name, img.GetHostname())
+				if err != nil {
+					glog.Error(err)
+					continue secretLoop
+				}
+
+				notaryToken, err := c.cr.GetContentTrustToken(registryToken, img.NameWithoutTag(), img.GetRegistryURL())
+				if err != nil {
+					glog.Error(err)
 					continue secretLoop
 				}
-				a.StringToAdmissionResponse(fmt.Sprintf("Deny %q, failed to get content trust information: %s", img.String(), err.Error()))
-				if _, ok := err.(store.ErrServerUnavailable); ok {
-					glog.Errorf("Trust server unavailable: %v", err)
-					return a.Flush()
+
+				var signers []Signer
+				if policy.Trust.SignerSecrets != nil {
+					// Generate a []Singer with the values for each signerSecret
+					signers = make([]Signer, len(policy.Trust.SignerSecrets))
+					for i, secretName := range policy.Trust.SignerSecrets {
+						signers[i], err = c.getSignerSecret(namespace, secretName.Name)
+						if err != nil {
+							a.StringToAdmissionResponse(fmt.Sprintf("Deny %q, could not get signerSecret from your cluster, %s", img.String(), err.Error()))
+							continue containerLoop
+						}
+					}
 				}
-				glog.Warningf("Failed to get trust information for %q: %v", img.String(), err)
-				continue containerLoop
+
+				// Get image digest
+				glog.Info("getting signed image...")
+				notaryURL, err := img.GetContentTrustURL()
+				if err != nil {
+					a.StringToAdmissionResponse(fmt.Sprintf("Trust Server/Image Configuration Error: %v", err.Error()))
+					continue containerLoop
+				}
+				digest, err := c.getDigest(notaryURL, img.NameWithoutTag(), notaryToken, img.GetTag(), signers)
+				if err != nil {
+					if strings.Contains(err.Error(), "401") {
+						continue secretLoop
+					}
+					a.StringToAdmissionResponse(fmt.Sprintf("Deny %q, failed to get content trust information: %s", img.String(), err.Error()))
+					if _, ok := err.(store.ErrServerUnavailable); ok {
+						glog.Errorf("Trust server unavailable: %v", err)
+						return a.Flush()
+					}
+					glog.Warningf("Failed to get trust information for %q: %v", img.String(), err)
+					continue containerLoop
+				}
+				glog.Infof("Mutation #: %d  Image name: %s", containerIndex+1, img.String())
```

Comment:
Please could you add the `containerType` variable into this log line, so that we know whether it's mutating the first of the Containers or the InitContainers?